diff --git a/src/include/libplacebo/colorspace.h b/src/include/libplacebo/colorspace.h
index 8de8b46..8ff9851 100644
--- a/src/include/libplacebo/colorspace.h
+++ b/src/include/libplacebo/colorspace.h
@@ -91,6 +91,7 @@ enum pl_color_levels {
 // The alpha representation mode.
 enum pl_alpha_mode {
     PL_ALPHA_UNKNOWN = 0,   // or no alpha channel present
+    PL_ALPHA_NONE = PL_ALPHA_UNKNOWN,
     PL_ALPHA_INDEPENDENT,   // alpha channel is separate from the video
     PL_ALPHA_PREMULTIPLIED, // alpha channel is multiplied into the colors
     PL_ALPHA_MODE_COUNT,
diff --git a/src/include/libplacebo/utils/libav_internal.h b/src/include/libplacebo/utils/libav_internal.h
index 58f7dbe..8328f4b 100644
--- a/src/include/libplacebo/utils/libav_internal.h
+++ b/src/include/libplacebo/utils/libav_internal.h
@@ -31,6 +31,7 @@
 #include <libavutil/pixdesc.h>
 #include <libavutil/display.h>
 #include <libavcodec/version.h>
+#include <libavformat/version.h>
 
 // Try importing <vulkan.h> dynamically if it wasn't already
 #if !defined(VK_API_VERSION_1_2) && defined(__has_include)
