diff --git a/lib/ldb/wscript b/lib/ldb/wscript
index 2a93426..b79c354 100644
--- a/lib/ldb/wscript
+++ b/lib/ldb/wscript
@@ -379,11 +379,7 @@ def build(bld):
                      install=False)
 
     bld.SAMBA_BINARY('ldb_key_value_sub_txn_tdb_test',
-                     bld.SUBDIR('ldb_key_value',
-                         '''ldb_kv_search.c
-                            ldb_kv_index.c
-                            ldb_kv_cache.c''') +
-                     'tests/ldb_key_value_sub_txn_test.c',
+                    'tests/ldb_key_value_sub_txn_test.c',
                      cflags='-DTEST_BE=\"tdb\"',
                      deps='cmocka ldb ldb_tdb_err_map',
                      install=False)
diff --git a/third_party/heimdal_build/wscript_build b/third_party/heimdal_build/wscript_build
index ebbf943..bd1c261 100644
--- a/third_party/heimdal_build/wscript_build
+++ b/third_party/heimdal_build/wscript_build
@@ -4,6 +4,7 @@ import os
 from samba_utils import SET_TARGET_TYPE
 from samba_autoconf import CURRENT_CFLAGS
 from samba_utils import LOAD_ENVIRONMENT, TO_LIST
+from samba_bundled import BUILTIN_LIBRARY
 
 def heimdal_path(p, absolute=False):
     hpath = os.path.join("../heimdal", p)
@@ -206,6 +207,15 @@ def HEIMDAL_LIBRARY(libname, source, deps, version_script=None, includes='', cfl
         orig_vscript_map = heimdal_path(version_script, absolute=False)
     else:
         orig_vscript_map = None
+    if BUILTIN_LIBRARY(bld, libname):
+        bld.SAMBA_SUBSYSTEM(libname, source,
+                            deps=deps,
+                            includes=includes,
+                            cflags=cflags,
+                            cflags_end=cflags_end,
+                            allow_warnings=allow_warnings)
+        return
+
     bld.SAMBA_LIBRARY(libname, source,
                       deps=deps,
                       includes=includes,
