diff --git a/buildtools/wafsamba/pkgconfig.py b/buildtools/wafsamba/pkgconfig.py
index 417bf03..e109231 100644
--- a/buildtools/wafsamba/pkgconfig.py
+++ b/buildtools/wafsamba/pkgconfig.py
@@ -60,6 +60,15 @@ def PKG_CONFIG_FILES(bld, pc_files, vnum=None, extra_name=None):
             t.env.LIB_RPATH = ''
         if vnum:
             t.env.PACKAGE_VERSION = vnum
+        LIBS_PRIVATE = []
+        REQUIRES_PRIVATE = []
+        for _target, _type in t.env['TARGET_TYPE'].items():
+            if _type == "SYSLIB" and 'LIB_' + _target.upper() in t.env:
+                LIBS_PRIVATE.append('-l%s' % _target)
+                if 'INCLUDES_' + _target.upper() in t.env and 'LIBPATH_' + _target.upper() in t.env and _target not in ['z']:
+                    REQUIRES_PRIVATE.append(_target)
+        t.env.LIBS_PRIVATE = ' '.join(LIBS_PRIVATE)
+        t.env.REQUIRES_PRIVATE = ' '.join(REQUIRES_PRIVATE)
         for v in [ 'PREFIX', 'EXEC_PREFIX', 'LIB_RPATH' ]:
             t.vars.append(t.env[v])
         bld.INSTALL_FILES(dest, target, flat=True, destname=base)
diff --git a/buildtools/wafsamba/wafsamba.py b/buildtools/wafsamba/wafsamba.py
index 8dff3d6..ea88533 100644
--- a/buildtools/wafsamba/wafsamba.py
+++ b/buildtools/wafsamba/wafsamba.py
@@ -322,7 +322,7 @@ def SAMBA_LIBRARY(bld, libname, source,
     if bld.env['ENABLE_RELRO'] is True:
         ldflags.extend(TO_LIST('-Wl,-z,relro,-z,now'))
 
-    features = 'c cshlib symlink_lib install_lib'
+    features = 'c cstlib'
     if pyext:
         features += ' pyext'
     if pyembed:
diff --git a/source3/libsmb/smbclient.pc.in b/source3/libsmb/smbclient.pc.in
index bcef2f2..2cf4df5 100644
--- a/source3/libsmb/smbclient.pc.in
+++ b/source3/libsmb/smbclient.pc.in
@@ -7,5 +7,7 @@ Name: smbclient
 Description: A SMB library interface
 Version: @PACKAGE_VERSION@
 Libs: @LIB_RPATH@ -L${libdir} -lsmbclient
+Libs.private: @LIBS_PRIVATE@
+Requires.private: @REQUIRES_PRIVATE@
 Cflags: -I${includedir}
 URL: http://www.samba.org/
diff --git a/third_party/heimdal/lib/hcrypto/camellia-ntt.h b/third_party/heimdal/lib/hcrypto/camellia-ntt.h
index cc0b70d..e79b1da 100644
--- a/third_party/heimdal/lib/hcrypto/camellia-ntt.h
+++ b/third_party/heimdal/lib/hcrypto/camellia-ntt.h
@@ -40,6 +40,10 @@ typedef unsigned char u8;
 
 typedef u32 KEY_TABLE_TYPE[CAMELLIA_TABLE_WORD_LEN];
 
+/* symbol renaming */
+#define Camellia_Ekeygen hc_Camellia_Ekeygen
+#define Camellia_EncryptBlock hc_Camellia_EncryptBlock
+#define Camellia_DecryptBlock hc_Camellia_DecryptBlock
 
 void Camellia_Ekeygen(const int keyBitLength,
 		      const unsigned char *rawKey,
