diff --git a/third_party/waf/waflib/Tools/python.py b/third_party/waf/waflib/Tools/python.py
index 4cf0019..812a8df 100644
--- a/third_party/waf/waflib/Tools/python.py
+++ b/third_party/waf/waflib/Tools/python.py
@@ -53,7 +53,18 @@ INST = '''
 Piece of Python code used in :py:class:`waflib.Tools.python.pyo` and :py:class:`waflib.Tools.python.pyc` for byte-compiling python files
 """
 
-DISTUTILS_IMP = ['from distutils.sysconfig import get_config_var, get_python_lib']
+DISTUTILS_IMP = ['''
+import sysconfig
+get_config_var = sysconfig.get_config_var
+def get_python_lib(plat_specific=False, standard_lib=False, prefix=None):
+	key = 'platlib' if plat_specific else 'purelib'
+	if standard_lib:
+		key = 'platstdlib' if plat_specific else 'stdlib'
+	vars = None
+	if prefix is not None:
+		vars = {'base': prefix, 'platbase': prefix}
+	return sysconfig.get_path(key, vars=vars)
+''']
 
 @before_method('process_source')
 @feature('py')
